R/workflows.R

```diff
@@ -0,0 +1,26 @@
+workflow_validate <- function(path, tasks) {
+  reports <- orderly::orderly_list(root = path)
+  missing_deps <- lapply(names(tasks), function(task) {
+    if (!(task %in% reports)) {
+      stop(sprintf("Task with name '%s' cannot be found.", task))
+    }
+    get_missing_dependencies(task, path, tasks)
+  })
+  list(missing_dependencies = setNames(missing_deps, names(tasks)))
+}
+
+get_missing_dependencies <- function(task, path, tasks) {
+  ## Not ideal that we build a graph of all dependencies then throw all
```

Comment:
I thought `max_depth` would achieve this for us but that just throws an error when that depth is reached. What do you think of adding another arg or changing the `max_depth` arg to return the tree just to that level of depth?

Reply:
Added a PR in orderly https://github.com/vimc/orderly/pull/278 to handle this and update code above to use `max_depth = 1`